setup.py

```diff
@@ -57,16 +57,23 @@
     extras_require={
         "system_certs": ["certifi>=2019.11.28"],
         "DEV": [
+            # certs
+            "certifi>=2019.11.28",
+            # code formatting/linting
             "black>=21.5b0",
             "flake8>=3.9.1",
             "pre-commit>=2.12.1",
+            # testing
             "pytest>=5.3.4",
             "pytest-mock>=3.6.1",
             "pytest-timeout>=1.4.2",
             "tox>=3.23.1",
+            # docs
             "pdoc3>=0.9.2",
-            "certifi>=2019.11.28",
             "numpydoc>=1.1.0",
+            "Sphinx==4.3.0",
+            "sphinx-rtd-theme==1.0.0",
```

Comment:
Any reason for fixing these versions?

Reply:
Just so that the output docs are stable (i.e. our machines build different docs from the same git commit because we have different versions of these installed).

We used to have this problem in core pach where we weren't careful about this, and then people were constantly panicking that e.g. "we accidentally forgot to build the protos in our last release, because when you run `make proto` in the tagged commit, the generated code is updated" (but actually it was just that the person who did the release had a different version of `protoc` installed). If the versions are fixed, then the output docs should only change when we change the docstrings or change the sphinx et. al. versions.

Reply:
Got it. So it's more to make sure that if the docs change, we know for a fact the change's source is the docstrings that were updated. Not that the sphinx version changed and creates the docs slightly differently.